### ISSUE TITLE
Rare segfault involving macroatoms

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#389]
This input file is causing occasional segfaults in the current dev version of python

[macro.pf.txt](https://github.com/agnwinds/python/files/2137518/macro.pf.txt)

This is a stellar wind model, in which the wind has been made to start at a distance outside of the star.  The segfault occurs in the routine bb_simple_heat which is called in this section of the code

![image](https://user-images.githubusercontent.com/2530901/41918867-d6a717ee-7922-11e8-8892-5f1557e1e802.png)

check_in_grid is returning 0, but xplasma2 is ending up undefined.  

two, which is an element of the wind, is element 0 of the spherical grid, which is NOT_IN_WIND, and points to NPLASMA 97.   This plasma cell has NWIND=-1 (because there is a many to 1 mapping for this type of NOT_IN_WIND cell.  It's the -1 that ultimately causes the trouble.

The original photon input to  caculate_ds  is in wind_cell 1, but smax (which has been given a nudge with DFUDGE) causes the two internal photons phot (current pos+smax) and p_now (somewhere in between) to be in cell 0

(This problem probably would not arise had the (historical) ping-ponging we often have between the plasma and wind structures.  Indeed there are some lines in the code above that say "code may be redundant")

Bottom line:  I think what is happening is that is that because of DFUDGE, it is possible to have a resonance that is just on the other side of the cell boundary, and if that next cell is not in the wind one can have problems when one tries to update the macro-atom estimators.  The easiest thing to do will be to trap this situation, and throw an error.

**Comment by kslong** [comment]
I believe that the fix I tried, which was to check that the volume in the cell was not zero worked, though it would be nice if @jhmatthews took a look.  Note also that bb_estimators_increment and bb_simple_heat do not have identical calls looks like the code was not cleaned up.  One calls the wind cell and the other calls the plasma cell.  Additionally, the logic still does seem a bit "redundant" as an earlier annotator  pointed out.
